### DISCUSSION TITLE
How to consider newlines as spaces

### DISCUSSION THREAD

**Discussion by lotodore** [yuin/goldmark#389] [General]
When using the goldmark renderer for gio-x https://github.com/gioui/gio-x/blob/main/markdown/markdown.go there is a problem handling newlines in markdown. For example, a newline before a link should be rendered as space before the link, but we are unsure how to achieve this. There is no space at the end of the text before the link, but also, we cannot always add a space before links, because links can be embedded in text without spaces.

Maybe someone could help how to render newlines as spaces before e.g. links?

**Comment by jmooring** [comment]
<details>
<summary>This code</summary>

```text
package main

import (
	"bytes"
	"fmt"

	"github.com/yuin/goldmark"
)

func main() {
	md := goldmark.New()

	input := `
text
[link](/foo)
`

	var buf bytes.Buffer
	if err := md.Convert([]byte(input), &buf); err != nil {
		panic(err)
	}

	fmt.Println(buf.String())

}
```
</details>

Produces:

```
<p>text
<a href="/foo">link</a></p>
```

Which is displayed as:

![image](https://user-images.githubusercontent.com/335669/232341751-48dd2339-74d6-4de3-9c43-73940294a90b.png)

**Comment by lotodore** [comment]
Thanks for your help @jmooring! I've debugged this and it helps to look at how this example works. Thanks for taking the time to post this example!
